adapters/33across/33across.go

```diff
@@ -135,16 +155,62 @@ func (a *TtxAdapter) MakeBids(internalRequest *openrtb.BidRequest, externalReque
 
 	for _, sb := range bidResp.SeatBid {
 		for i := range sb.Bid {
+			var bidExt bidExt
+			var bidType openrtb_ext.BidType
+
+			if err := json.Unmarshal(sb.Bid[i].Ext, &bidExt); err != nil {
+				bidType = openrtb_ext.BidTypeBanner
+			} else {
+				bidType = getBidType(bidExt)
+			}
+
 			bidResponse.Bids = append(bidResponse.Bids, &adapters.TypedBid{
 				Bid:     &sb.Bid[i],
-				BidType: "banner",
+				BidType: bidType,
 			})
 		}
 	}
 	return bidResponse, nil
 
 }
 
+func validateVideoParams(video *openrtb.Video, prod string) (*openrtb.Video, error) {
+	videoCopy := video
+	if videoCopy.W == 0 ||
+		videoCopy.H == 0 ||
+		videoCopy.Protocols == nil ||
+		videoCopy.MIMEs == nil ||
+		videoCopy.PlaybackMethod == nil {
+
+		return nil, &errortypes.BadInput{
+			Message: "One or more invalid or missing video field(s) w, h, protocols, mimes, playbackmethod",
+		}
+	}
+
+	if videoCopy.Placement == 0 {
+		videoCopy.Placement = 2
+	}
+
+	if prod == "instream" {
+		videoCopy.Placement = 1
+
+		if videoCopy.StartDelay == nil {
+			videoCopy.StartDelay = openrtb.StartDelay.Ptr(0)
+		}
+	}
+
+	return videoCopy, nil
+}
+
+func getBidType(ext bidExt) openrtb_ext.BidType {
+	if ext.Ttx.MediaType == "video" {
+		return openrtb_ext.BidTypeVideo
+	}
+
+	return openrtb_ext.BidTypeBanner
```

Comment:
Please add test coverage for this assume banner if not a video mediatype branch.